sql/wsrep_mysqld.cc

```diff
@@ -2279,23 +2286,15 @@ static my_bool kill_remaining_threads(THD *thd, THD *caller_thd)
 
 void wsrep_close_client_connections(my_bool wait_to_end, THD* except_caller_thd)
 {
+  /* Clear thread cache */
+  kill_cached_threads++;
+  flush_thread_cache();
+  
```

Comment:
Do you really need to care about it? FWICS at least applier threads are never added to the thread cache.